plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -561,19 +561,16 @@ public Map<String, PartitionStatistics> getPartitionStatistics(Table table, List
 
         Map<HivePartitionName, PartitionStatistics> missingPartitionsStats = loadPartitionsColumnStatistics(table, partitionsByName, missingPartitions.build());
         // merge missing stats into the cached stats
-        Map<HivePartitionName, AtomicReference<PartitionStatistics>> partitionStatisticsCacheMap = partitionStatisticsCache.asMap();
         missingPartitionsStats.forEach((partition, stats) -> {
             // make sure cache is not empty to avoid putIfAbsent call
-            uncheckedCacheGet(partitionStatisticsCache, partition, AtomicReference::new);
-            partitionStatisticsCacheMap.computeIfPresent(partition, (ignored, statsHolder) -> {
+            AtomicReference<PartitionStatistics> currentReference = uncheckedCacheGet(partitionStatisticsCache, partition, AtomicReference::new);
```

Comment:
still there is a race between `loadPartitionsColumnStatistics` and `invalidation` (similar as for table stats)

Reply:
moved to the new `getAll` and fixed